internal/lru/lru.go

```diff
@@ -100,9 +139,20 @@ func (c *Cache) Remove(key string) bool {
 
 // RemoveOldest removes the oldest item from the cache.
 func (c *Cache) RemoveOldest() {
-	if c.cache == nil {
-		return
-	}
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.removeOldest()
```

Comment:
Rename to removeOldestLocked, possily apply the pattern in other places too if needed